utils/file.go

```diff
@@ -95,7 +95,33 @@ func CopyFile(ctx context.Context, src, dst string, mode os.FileMode) (err error
 		}
 	}
 
-	return CopyFileContents(ctx, src, dst, mode)
+	err = os.MkdirAll(filepath.Dir(dst), 0o750)
+	if err != nil {
+		return err
+	}
+
+	out, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+
+	// Change file ownership to user running Containerlab instead of effective UID
+	err = SetUIDAndGID(dst)
+	if err != nil {
+		return err
+	}
+
+	err = out.Chmod(mode)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		// should only err on repeated calls to close anyway
+		_ = out.Close()
+	}()
+
+	return CopyFileContents(ctx, src, out)
```

Comment:
looks like you removed this logic from the CopyFileContents func presumably to make it do less while being more closer to the function name?

I am fine with that, but I wonder if all the mechanics of ensuring that dst exists and uid/gid are set correctly should be factored out in a func?

Because right now this code piece is copied here in the `CopyFile` func as well as in the `core/config.go`